pkg/provisioning/service.go

```diff
@@ -234,51 +242,82 @@ func (s *Service) applyConnectorStates(ctx context.Context, pl *pipeline.Instanc
 	return nil
 }
 
-func (s *Service) deletePipeline(ctx context.Context, pl *pipeline.Instance, config PipelineConfig) error {
+func (s *Service) deletePipeline(ctx context.Context, r *rollback.R, pl *pipeline.Instance, config PipelineConfig, connStates map[string]func(context.Context) error) error {
 	var err error
 	// remove pipeline processors
-	for k := range config.Processors {
+	for k, v := range config.Processors {
 		_, err = s.pipelineService.RemoveProcessor(ctx, pl, k)
 		if err != nil {
 			return err
 		}
+		s.rollbackRemoveProcessorFromPipeline(ctx, r, pl, k)
+
 		err = s.processorService.Delete(ctx, k)
 		if err != nil {
 			return err
 		}
+		s.rollbackDeleteProcessor(ctx, r, pl.ID, processor.ParentTypePipeline, k, v)
 	}
+
+	r.Append(func() error {
+		err := s.applyConnectorStates(ctx, pl, connStates)
+		if err != nil {
+			return err
+		}
+		if pl.Status == pipeline.StatusRunning {
```

Comment:
Given that we first stop the pipeline this will _probably_ not be `StatusRunning`. We might want to move the stopping of the pipeline into this function and if we need to stop it also add the rollback to start the pipeline again.